pandas/io/formats/style.py

```diff
@@ -1060,12 +1061,76 @@ def hide_columns(self, subset) -> Styler:
         -------
         self : Styler
         """
-        subset = non_reducing_slice(subset)
-        hidden_df = self.data.loc[subset]
-        hcols = self.columns.get_indexer_for(hidden_df.columns)
-        # error: Incompatible types in assignment (expression has type
-        # "ndarray", variable has type "Sequence[int]")
-        self.hidden_columns = hcols  # type: ignore[assignment]
+        return self.hide_values(subset)
+
+    def hide_values(self, subset, axis: Axis = "columns", show: bool = False) -> Styler:
+        """
```

Comment:
i wouldn't make this user visible instead prefer hide_index, hide_columns (this can be the impl)